[Stream] Added docs for VOD clipping (#5400)
cloudflare/cloudflare-docs: content/stream/edit-manage-videos/edit-videos/video-clipping.md

---
pcx-content-type: how to
title: Clip videos
---

# Clip videos

> With video clipping, also referred to as "trimming" or changing the length of the video, you can change the start and end points of a video so viewers only see a specific "clip" of the video. For example, if you have a 20 minute video where the first five minutes include event setup and waiting for the event to start, you can clip the video to remove the first five minutes and start when the event actually begins.

Reviewer: ```suggestion
With video clipping, also referred to as "trimming" or changing the length of the video, you can change the start and end points of a video so viewers only see a specific clip of the video. For example, if you have a 20 minute video where the first five minutes include event setup and waiting for the event to start, you can clip the video to remove the first five minutes and start when the event begins.
```

## Prerequisites

Before you can clip a video, you will need an API token. For more information on creating an API token, refer to [Creating API tokens](/api/tokens/create/).

## Required parameters

> To clip your video, determine the start and end times you want to use from the existing video to create the new video. Use that `videoUID` and the start end times to make your request.

Reviewer: The second sentence feel awkward, does `videoUID` have a noun to describe it? or is that supposed to be the?

Author: Ah, good catch. it's definitely supposed to be "the" and not "that."

    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "meta": {
      "name": "overriding-filename-clip.mp4"
    }
}'
```

You can check whether your video is ready to play after selecting your account from the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/stream). While the clipped video processes, the video status response displays **Queued**. When the clipping process is complete, the video status changes to **Ready** and displays the new name of the clipped video and the new duration.

To receive a notification when your video is done processing and ready to play, you can [subscribe to webhook notifications](/stream/edit-manage-videos/manage-video-library/using-webhooks/).

## Set video name

When you clip a video, you can also specify a new name for the clipped video. In the example below, the `name` field indicates the new name to use for the clipped video.

```bash
---
header: Example: Specify a custom name
highlight: [6]
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "meta": {
      "name": "overriding-filename-clip.mp4"
    }
}
```

> When the video has been clipped and processed, the video with the new name displays in your Cloudflare dashboard in the list videos.

Reviewer: ```suggestion
When the video has been clipped and processed, your newly named video displays in your Cloudflare dashboard in the list videos.
```

Reviewer: Is List Videos a section in the dash? Seems like it may need to be capitalized and bolded?

Author: Fixed! "List videos" isn't a section of the dash. The list of videos is actually unnamed and just a section at the bottom of the page.

## Add a watermark

> You can clip a video and also add a custom watermark. For more information on watermarks and uploading a watermark profile, refer to [Apply watermarks](/stream/edit-manage-videos/edit-videos).

Reviewer: ```suggestion
You can also add a custom watermark to your video. For more information on watermarks and uploading a watermark profile, refer to [Apply watermarks](/stream/edit-manage-videos/edit-videos).
```

Reviewer: Can you only add a watermark if you clipped a video? Or can you have the video as is and add a watermark? If the latter, proposing a change.

Author: It's the latter. Fixed

```bash
---
header: Example: Clip a video, set a new video name, and apply a watermark
highlight: [5,6,9]
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "watermark": {
        "uid": "4babd675387c3d927f58c41c761978fe"
    },
    "meta": {
      "name": "overriding-filename-clip.mp4"
    }
}
```

## Require signed URLs

When clipping a video, you can make a video private and accessible only to certain users by [requiring a signed URL](/stream/viewing-videos/securing-your-stream/).

```bash
---
header: Example: Clip a video and require signed URLs
highlight: [5]
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "requireSignedURLs": true,
    "meta": {
      "name": "signed-urls-demo.mp4"
    }
}
```

> After the video clipping is complete, you can open the Cloudflare dashboard and video list to locate your video. When you select the video, the **Settings** tab displays a checkmark next to **Require Signed URLs**.

Reviewer: ```suggestion
After the video clipping is complete, you can open the Cloudflare dashboard and video list to locate your video. When you select the video, **Settings** displays a checkmark next to **Require Signed URLs**.
```

Reviewer: potentially unsuitable amendment

Author: I'm gonna leave this one as is because there are a few different tabs up top and it's not necessarily a field that displays. Good callout though

## Specify thumbnail percentages

> You can clip a video and also specify a thumbnail for your video using a percentage value. To convert the thumbnail's timestamp from seconds to a percentage, divide the timestamp you want to use by the total duration of the video. For more information about thumbnails, refer to [Display thumbnails](/stream/edit-manage-videos/edit-videos/displaying-thumbnails).

Reviewer: Same comment as above.

> You can clip a video and also specify a thumbnail for your video using a percentage value. To convert the thumbnail's timestamp from seconds to a percentage, divide the timestamp you want to use by the total duration of the video. For more information about thumbnails, refer to [Display thumbnails](/stream/edit-manage-videos/edit-videos/displaying-thumbnails).

Reviewer: ```suggestion
You can specify a thumbnail for your video using a percentage value. To convert the thumbnail's timestamp from seconds to a percentage, divide the timestamp you want to use by the total duration of the video. For more information about thumbnails, refer to [Display thumbnails](/stream/edit-manage-videos/edit-videos/displaying-thumbnails).
```

Author: Fixed

---
pcx-content-type: how to
title: Clip videos
---

# Clip videos

> With video clipping, also referred to as "trimming" or changing the length of the video, you can change the start and end points of a video so viewers only see a specific "clip" of the video. For example, if you have a 20 minute video where the first five minutes include event setup and waiting for the event to start, you can clip the video to remove the first five minutes and start when the event begins.

Reviewer 2: Need to change the example to _not_ be about live videos.

Author: Updated the example

{{<Aside type="note" header="Note:">}}

> At this time, video clipping only works with videos on demand. 

Reviewer 2: Let's avoid calling them "videos on demand".

Think it might be simpler to say what is _not_ supported, rather than what is _only_ supported, ex: "video clipping does not work with recordings of live streams"

Author: Fixed

## Required parameters

To clip your video, determine the start and end times you want to use from the existing video to create the new video. Use the `videoUID` and the start end times to make your request.

```bash
---
header: Required parameters
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 20,
    "endTimeSeconds": 40
}
```

- **`clippedFromVideoUID`**: The unique identifier for the video used to create the new, clipped video.
- **`startTimeSeconds`**: The timestamp from the existing video that indicates when the new video begins.
- **`endTimeSeconds`**: The timestamp from the existing video that indicates when the new video ends.
</br></br>

```bash
---
header: Example: Clip a video
highlight: [5,6,7]
---
curl --location --request POST 'https://api.staging.cloudflare.com/client/v4/accounts/<YOUR_ACCOUND_ID_HERE>/stream/clip' \
--header 'Authorization: Bearer <YOUR_TOKEN_HERE>' \
--header 'Content-Type: application/json' \
--data-raw '{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "meta": {

>       "name": "overriding-filename-clip.mp4"

Reviewer 2: Using a filename here makes it look like there's some important connection between the name of the video and the original filename, while in reality, it's just that the filename is the default name for a video, if none is provided.

If someone is renaming a video, as is the case here, it seems like we could use a normal, title-cased name, like "<Video Name> [clipped version]" or similar.

Not a blocking comment to ship this change.

Reviewer 2: Ex: once a video is uploaded to stream, the file type of the original (ex: mp4) should always be irrelevant to the person or developer using Stream — once uploaded, it's just a video in Stream, that _just works_.

Author: Removed name to only focus on required parameters

    }
}'
```

You can check whether your video is ready to play after selecting your account from the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/stream). While the clipped video processes, the video status response displays **Queued**. When the clipping process is complete, the video status changes to **Ready** and displays the new name of the clipped video and the new duration.

To receive a notification when your video is done processing and ready to play, you can [subscribe to webhook notifications](/stream/edit-manage-videos/manage-video-library/using-webhooks/).

> ## Set video name

Reviewer 2: If we have an explicit section for this here, then it seems like the above example request should _not_ include a name in it, and should _only_ included the parameters that are truly required.

Author: Removed the name from the previous request

```bash
---
header: Example: Clip a video, set a new video name, and apply a watermark
highlight: [5,6,9]
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "watermark": {
        "uid": "4babd675387c3d927f58c41c761978fe"
    },
    "meta": {
      "name": "overriding-filename-clip.mp4"
    }
}
```

## Require signed URLs

When clipping a video, you can make a video private and accessible only to certain users by [requiring a signed URL](/stream/viewing-videos/securing-your-stream/).

```bash
---
header: Example: Clip a video and require signed URLs
highlight: [5]
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "requireSignedURLs": true,
    "meta": {
      "name": "signed-urls-demo.mp4"
    }
}
```

After the video clipping is complete, you can open the Cloudflare dashboard and video list to locate your video. When you select the video, the **Settings** tab displays a checkmark next to **Require Signed URLs**.

> ## Specify thumbnail percentages

Reviewer 2: I don't think this title is accurate — the job that we're explaining is something like "you can set what thumbnail image to display" — the percentage is just the _means_ to that end, the way you define the timestamp to generate the thumbnail.

Author: ah, good callout. Fixed

You can also specify a thumbnail for your video using a percentage value. To convert the thumbnail's timestamp from seconds to a percentage, divide the timestamp you want to use by the total duration of the video. For more information about thumbnails, refer to [Display thumbnails](/stream/edit-manage-videos/edit-videos/displaying-thumbnails).

```bash
---
header: Example: Clip a video with a thumbnail generated at the 50% mark
highlight: [5]
---
{
    "clippedFromVideoUID": "0ea62994907491cf9ebefb0a34c1e2c6",
    "startTimeSeconds": 10,
    "endTimeSeconds": 15,
    "thumbnailTimestampPct": 0.5,
    "meta": {
      "name": "thumbnail_percentage.mp4"
    }
}
```
## Example response

> The example response below highlights the optional parameters mentioned above. 

Reviewer 2: Once the API docs changes are live, it seems like this section could just link directly there, since a full request/response example is included for each API endpoint. cc @Schachte
